pkg/chunk/purger/purger.go

```diff
@@ -83,18 +82,20 @@ type deleteRequestWithLogger struct {
 	logger log.Logger // logger is initialized with userID and requestID to add context to every log generated using this
 }
 
-// Config holds config for DataPurger
+// Config holds config for Purger
 type Config struct {
-	Enable          bool   `yaml:"enable"`
-	NumWorkers      int    `yaml:"num_workers"`
-	ObjectStoreType string `yaml:"object_store_type"`
+	Enable                              bool          `yaml:"enable"`
+	NumWorkers                          int           `yaml:"num_workers"`
+	ObjectStoreType                     string        `yaml:"object_store_type"`
+	AllowDeleteRequestCancellationUntil time.Duration `yaml:"allow_delete_request_cancellation_until"`
```

Comment:
What about `CancelPeriodDuration`? It is shorter and I think it describes the intent well.

Reply:
Or `RequestCancellationPeriod`.

Reply:
Well, naming is hard. How about `DeleteRequestCancellableUntil`? I wanted to add `Delete` word to the flag to make it clearer. I can drop it if you think that is too much for a flag and people should infer from it or refer the docs to understand what type of request we are talking about.
What do you think?

Reply:
> Well, naming is hard. How about `DeleteRequestCancellableUntil`? I wanted to add `Delete` word to the flag to make it clearer. I can drop it if you think that is too much for a flag and people should infer from it or refer the docs to understand what type of request we are talking about.
> What do you think?

I'd like to get rid of "Until" part as that implies static timestamp to me, and prefer to use Period or Duration instead. Also in Go world, people spell Cancelled as "Canceled", we can be consistent.

So: `DeleteRequestCancelationPeriod` ?

Reply:
Or slightly shorter: `DeleteRequestCancelPeriod`

Reply:
Changed it to `DeleteRequestCancelPeriod`. Thanks!